src/frontend/js/api/joanie.ts

```diff
@@ -464,6 +467,16 @@ const API = (): Joanie.API => {
           return fetchWithJWT(buildApiUrl(ROUTES.courses.products.get, filters)).then(checkStatus);
         },
       },
+      orders: {
+        get: async (filters?: Joanie.CourseOrderResourceQuery) => {
+          if (!filters || !filters.course_id) {
+            throw new Error('A course_id is required to fetch a course order');
+          }
```

Comment:
Instead to throw an error, we could just update the function signature.

```suggestion
        get: async (filters: Joanie.CourseOrderResourceQuery) => {
```

Reply:
`filters` have to be optional from our api interface :
```ts
export interface ApiResourceInterface<
  TData extends Resource,
  TResourceQuery extends ResourcesQuery = ResourcesQuery,
> {
  get: (filters?: TResourceQuery) => any;
  create?: (payload: any) => Promise<TData>;
  update?: (payload: any) => Promise<TData>;
  delete?: (id: TData['id']) => Promise<void>;
}
```

Reply:
Ok all this topic about API Interface typing and required filter should not be adressed right now. Can you juste create an issue to not forget it.